src/panels/lovelace/common/compute-card-size.ts

```diff
@@ -1,10 +1,17 @@
+import { promiseTimeout } from "../../../common/util/promise-timeout";
 import { LovelaceCard, LovelaceHeaderFooter } from "../types";
 
 export const computeCardSize = (
   card: LovelaceCard | LovelaceHeaderFooter
 ): number | Promise<number> => {
   if (typeof card.getCardSize === "function") {
-    return card.getCardSize();
+    try {
+      return promiseTimeout(500, card.getCardSize()).catch(
+        () => 1
+      ) as Promise<number>;
+    } catch (_e: any) {
+      return 1;
+    }
```

Comment:
Why the double catch?

Reply:
It's been a while, but as I recall it things work differently whether getCardSize returns a promise or not.
It could be that without the `.catch()` a delayed exception falls through to the caller of `computeCardSize` or something... I can take a second look at it. I should still have my reproduction setup around somewhere.

Reply:
we don't await the promise, so the catch wont work